apps/events/models.py

```diff
@@ -909,14 +909,42 @@ def delete(self):
 
         super(Attendee, self).delete()
 
-    def get_payment_deadline(self):
+    @property
+    def payment_relations(self):
+        # Importing here to prevent circular dependencies
+        from apps.payment.models import PaymentRelation
+
+        payment = self.event.payment()
+        if payment:
+            payment_relations = PaymentRelation.objects.filter(payment=payment, user=self.user)
+            return payment_relations
+        return None
+
+    @property
+    def has_paid(self):
+        """
+        Useful for checking if the user has paid, and if anything potentially needs refunding.
+        :return: If the user has paid anything for the event.
+        """
+        if self.payment_relations:
+            non_refunded_payment_relations = self.payment_relations.filter(refunded=False)
+            return len(non_refunded_payment_relations) != 0
+        return False
+
+    @property
+    def payment_delays(self):
```

Comment:
I would like that you add type-annotations, for easier use later on

Reply:
While I would agree that it could be nice, I think that is a bit outside the scope of this change.